### ISSUE TITLE
dart2js renames some names without reason

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#3993]
Why is local name 'length' renamed to 'length$' in the following code?
It seem unnecessary.

$$.HashMapImplementation = {&quot;&quot;:
&nbsp;...
&nbsp;forEach$1: function(f) {
&nbsp;&nbsp;var length$ = $.get$length(this._keys);
&nbsp;&nbsp;if (typeof length$ !== 'number') return this.forEach$1$bailout(1, f, length$);
&nbsp;&nbsp;for (var i = 0; i &lt; length$; ++i) {
&nbsp;&nbsp;&nbsp;&nbsp;var key = $.index(this._keys, i);
&nbsp;&nbsp;&nbsp;&nbsp;!(key == null) &amp;&amp; !(key &equals;&equals;&equals; $.CTC2) &amp;&amp; f.$call$2(key, $.index(this._values, i));
&nbsp;&nbsp;}
&nbsp;},

**Comment by sigmundch** [comment]
I don't believe this to be an issue anymore - @rakudrama  please reopen if I'm mistaken.

**Comment by rakudrama** [comment]
We still needlessly add `$` signs to locals that have certain names, e.g. $length.
